jujugui/static/gui/src/app/components/terminal/terminal.js

```diff
@@ -78,17 +83,58 @@ class Terminal extends React.Component {
       }
       // Terminado sends a "disconnect" message when the process it's running
       // exits. When we receive that, we close the terminal.
-      if (resp['0'] === 'disconnect') {
-        this.close();
+      switch(resp[0]) {
+        case 'disconnect':
+          this.close();
+          break;
+        case 'setup':
```

Comment:
When is this sent by terminado? Please add a comment.

Reply:
done